### PR TITLE
DASHBUILDE-75: Data set API support for non-CDI containers

### PR DESCRIPTION
**Pull request by dgutierr**
This PR is a refactor of the data set subsystem in order to make it runnable under non-CDI containers just as any other plain java library. This includes the ability to register data set definitions as well as to execute data lookup calls against those definitions. This makes possible to embed the data set engine into other java applications and runtimes without the need of a CDI environment.

(This non-cdi support has been required by the KIE execution server)

Instructions on the Dataset API usage can be found [here](https://github.com/dgutierr/dashbuilder/tree/non-cdi-core/dashbuilder-backend/dashbuilder-dataset-core). I've also prepared a [project template](https://github.com/dgutierr/datasets-sample-project) that can be helpful for those who want to play a bit with the API (do not forget to build dashbuilder from this PR first, otherwise the template project build will fail).

### CODE REVIEW

**Comment by mswiderski** [comment]
looks perfect to me. Good and clear separation and the code module (api and shared) has very few dependencies and no cdi deps for core.

+1

File: dashbuilder-backend/dashbuilder-dataset-cdi/pom.xml [hunk 1]
```diff
@@ -0,0 +1,197 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright 2014 JBoss Inc
```
Comment (psiroky) [inline dashbuilder-backend/dashbuilder-dataset-cdi/pom.xml]: The header should not contain references to JBoss Inc (it does not exist for some time now), see https://github.com/droolsjbpm/droolsjbpm-build-bootstrap/blob/master/ide-configuration/LICENSE-ASL-2.0-HEADER.txt

File: dashbuilder-backend/dashbuilder-dataset-cdi/src/main/java/org/dashbuilder/Bootstrap.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (C) 2015 JBoss Inc
```
Comment (psiroky) [inline dashbuilder-backend/dashbuilder-dataset-cdi/src/main/java/org/dashbuilder/Bootstrap.java]: The header should not contain references to JBoss Inc (it does not exist for some time now), see https://github.com/droolsjbpm/droolsjbpm-build-bootstrap/blob/master/ide-configuration/LICENSE-ASL-2.0-HEADER.txt

File: dashbuilder-backend/dashbuilder-dataset-cdi/src/main/java/org/dashbuilder/dataset/DataSetDefDeployerCDI.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (C) 2014 JBoss Inc
```
Comment (psiroky) [inline dashbuilder-backend/dashbuilder-dataset-cdi/src/main/java/org/dashbuilder/dataset/DataSetDefDeployerCDI.java]: The header should not contain references to JBoss Inc (it does not exist for some time now), see https://github.com/droolsjbpm/droolsjbpm-build-bootstrap/blob/master/ide-configuration/LICENSE-ASL-2.0-HEADER.txt

**Comment by dgutierr** [comment]
We agreed with @mbiarnes  to hold on the headers replacement until this PR is fixed in order to avoid too many conflicts. That means it's not necessary to set the headers correctly as part of this PR since that will be entirely fixed later on.

**Pull request closed without merging**